docs/reference/analysis/tokenfilters/synonym-tokenfilter.asciidoc

```diff
@@ -181,3 +185,40 @@ error.
 If you need to build analyzers that include both multi-token filters and synonym
 filters, consider using the <<analysis-multiplexer-tokenfilter,multiplexer>> filter,
 with the multi-token filters in one branch and the synonym filter in the other.
+
+=== Updateability of search time synonyms
+
+Synonym filters that are used in <<search analyzers,search-analyzer>> can be marked
+as updateable using the `updateable` flag:
+
+[source,js]
+--------------------------------------------------
+PUT /test_index
+{
+    "settings": {
+        "index" : {
+            "analysis" : {
+                "analyzer" : {
+                    "synonym" : {
+                        "tokenizer" : "whitespace",
+                        "filter" : ["synonym"]
+                    }
+                },
+                "filter" : {
+                    "synonym" : {
+                        "type" : "synonym",
+                        "synonyms_path" : "analysis/synonym.txt",
+                        "updateable" : true
+                    }
+                }
+            }
+        }
+    }
+}
+--------------------------------------------------
+// CONSOLE
+
+Using the <<analyzer reload API,indices-reload-analyzers>>, you can trigger reloading of the
+synonym definition. The contents of the configured synonyms file will be reloaded and the
+synonyms definition the filter uses will be updated. Note that if you trying to use the above
+analyzer as an index analyzer will result in an error.
```

Comment:
jimczi on #42865: I would extract the last sentence in a NOTE in order to make sure that users are aware of the limitation.